interview-questions.md

```diff
@@ -7,44 +7,112 @@ Then, research the question to expand on your answer. Even if you feel you have
+    - rails g resource Student name:string cohort:string
+  2. rails generate controller - creates a new controller
+    - rails generate controller NAME [action action] [options]
+  3. rails generate migration - Migrations are used to modify the shape of an existing database - this cn include adding or deleting columns, changing the name of the column, or changing the data type of a column.
 
   Researched answer:
 
+    1. rails generate resource - does everything generating a model does, but it also creates a resource in the routes.rb file and an empty folder in the views. Example: resource name along with the table column names and types. Everything is set.  NO SETTING ROUTES. Thank the Rails overlords. 
+    - rails g resource Student name:string cohort:string
+   2. rails generate controller - creates a new controller
+    - rails generate controller Cohort [action action] [options]
+    - For example, to generate a controller for a Cohort model, with #index and #show actions you would run rails generate controller cohorts index show. Also creates folder inside app/views/, containing the two templates for the controller's actions. 
+   3. rails generate migration - Migrations are used to modify the shape of an existing database - adding or deleting columns, changing the name or changing the data type of a column. Migration names start with an action that describes the purpose of the migration (i.e. create, add, remove, etc). The migration name ends with the name of the table (add_item_to_list)
+
 
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-action: "GET"    location: /students          
+action: "GET"    location: /students   - index method
 
-action: "POST"   location: /students       
+action: "POST"   location: /students  -  create method    
 
-action: "GET"    location: /students/new
+action: "GET"    location: /students/new - new method - form
 
-action: "GET"    location: /students/2  
+action: "GET"    location: /students/2  - show (show one piece - show student id: 2
 
-action: "GET"    location: /students/2/edit    
+action: "GET"    location: /students/2/edit   - edit student id: 2 (form)
 
-action: "PATCH"  location: /students/2      
+action: "PATCH"  location: /students/2   - update - modifying student id: 2
 
-action: "DELETE" location: /students/2      
+action: "DELETE" location: /students/2      destroy - remove student id: 2
 
 
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
+- As a developer, I can create a class called Task.
```

Comment:
These are all great developer stories.  User stories are usually geared more toward what the user experience will be vs what the developer will be creating.